test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/RazorGenerateIntegrationTest.cs

```diff
@@ -15,9 +16,19 @@ public class RazorGenerateIntegrationTest : MSBuildIntegrationTestBase
 
         [Fact]
         [InitializeTestProject("SimpleMvc")]
-        public async Task RazorGenerate_Success_GeneratesFilesOnDisk()
+        public Task RazorGenerate_UsingDotNetMSBuild_Success_GeneratesFilesOnDisk()
+            => RazorGenerate_Success_GeneratesFilesOnDisk(MSBuildProcessKind.Dotnet);
+
+        [ConditionalFact]
+        [OSSkipCondition(OperatingSystems.Linux)]
+        [OSSkipCondition(OperatingSystems.MacOSX)]
+        [InitializeTestProject("SimpleMvc")]
+        public Task RazorGenerate_UsingDesktopMSBuild_Success_GeneratesFilesOnDisk()
+            => RazorGenerate_Success_GeneratesFilesOnDisk(MSBuildProcessKind.Desktop);
+
+        private async Task RazorGenerate_Success_GeneratesFilesOnDisk(MSBuildProcessKind msBuildProcessKind)
         {
-            var result = await DotnetMSBuild(RazorGenerateTarget);
+            var result = await DotnetMSBuild(RazorGenerateTarget, msBuildProcessKind: msBuildProcessKind);
```

Comment:
Can we limit this to just `Build_SimpleMvc_CanBuildSuccessfully`? I don't want to do cross product testing, and we have no reason to believe that our code will behave differently across different MSBuild kinds.

Reply:
Sure